src/components/MultipieceModal/MultipieceModal.test.js

```diff
@@ -0,0 +1,240 @@
+
+describe('MultipieceModal', () => {
+  const testIds = {
+    multipieceModal: 'multipieceModal',
+    multipieceModalFooter: 'multipieceModalFooter',
+    multipieceModalConfirmButton: 'multipieceModalConfirmButton',
+    multipieceModalCancelButton: 'multipieceModalCancelButton',
+    numberOfPieces: 'numberOfPieces',
+    descriptionOfPieces: 'descriptionOfPieces',
+    numberOfMissingPieces: 'numberOfMissingPieces',
+    descriptionOfmissingPieces: 'descriptionOfmissingPieces',
+  };
+  const labelIds = {
+    multipieceModalLabel: 'ui-checkin.multipieceModal.label',
+    multipieceModalCheckIn: 'ui-checkin.multipieceModal.checkIn',
+    multipieceModalCancel: 'ui-checkin.multipieceModal.cancel',
+    multipieceModalMessage: 'ui-checkin.multipieceModal.message',
+    itemNumberOfPieces: 'ui-checkin.multipieceModal.item.numberOfPieces',
+    itemDescriptionOfPieces: 'ui-checkin.multipieceModal.item.descriptionOfPieces',
+    itemNumberOfMissingPieces: 'ui-checkin.multipieceModal.item.numberOfMissingPieces',
+    itemDescriptionOfMissingPieces: 'ui-checkin.multipieceModal.item.descriptionOfmissingPieces',
+  };
+  const testOpen = false;
+  const testItem = {
+    title: 'testTitle',
+    barcode: 'testBarcode',
+    materialType: {
+      name: 'testName',
+    },
+  };
+  const testOnClose = jest.fn();
+  const testOnConfirm = jest.fn();
+  const testDefaultProps = {
+    onClose: testOnClose,
+    onConfirm: testOnConfirm,
+    item: testItem,
+    open: testOpen,
+  };
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('with default props', () => {
+    beforeEach(() => {
+      render(<MultipieceModal {...testDefaultProps} />);
+    });
+
+    it('should render multipiece "Modal" component', () => {
+      componentPropsCheck(Modal, testIds.multipieceModal, {
+        id: 'multipiece-modal',
+        size: 'small',
+        dismissible: true,
+        label: labelIds.multipieceModalLabel,
+      }, true);
+    });
+
+    it('should render multipiece "ModalFooter" component', () => {
+      componentPropsCheck(ModalFooter, testIds.multipieceModalFooter, {}, true);
+    });
+
+    it('should render multipiece modal confirm "Button" component', () => {
+      componentPropsCheck(Button, testIds.multipieceModalConfirmButton, {
+        buttonStyle: 'primary',
+      }, true);
+    });
+
+    it('should render multipiece modal confirm "Button" component label', () => {
+      expect(getById(testIds.multipieceModalConfirmButton).getByText(labelIds.multipieceModalCheckIn)).toBeVisible();
+    });
+
+    it('should call onConfirm callback when click on confirm button', () => {
+      expect(testOnConfirm).not.toBeCalled();
+
+      fireEvent.click(screen.getByTestId(testIds.multipieceModalConfirmButton));
+
+      expect(testOnConfirm).toHaveBeenCalled();
+    });
+
+    it('should render multipiece modal cancel "Button" component', () => {
+      componentPropsCheck(Button, testIds.multipieceModalCancelButton, {}, true);
```

Comment:
The same with props checking here

Reply:
We have multiple places where Button component is used. We want to check that specific Button(in this case cancel button) is called